packages/core/src/external/mpi/patient-blocker.ts

```diff
@@ -0,0 +1,20 @@
+import { GenderAtBirth, PatientDataMPI } from "./patient";
+
+export type PatientBlock = {
+  cxId?: string;
+  facilityIds?: string[];
+  data?: {
+    dob?: string;
+    genderAtBirth?: GenderAtBirth;
+    firstNameInitial?: string;
+    lastNameInitial?: string;
+  };
+};
+
+export abstract class PatientBlocker {
+  abstract block(data: PatientBlock): Promise<PatientDataMPI[]>;
+}
```

Comment:
Can we use an approach similar to `PatientLoader` (there's also a `PatientUpdater`)?
- `PatientLoader` - interface
- `PatientLoaderMetriportAPI` - implementation that calls the Metriport API to return Patient-related data
- `PatientLoaderLocal` - implementation that loads Patient-related data directly from the DB

We could use those existing interfaces/classes for this (finding/searching Patients), could even rename to `PatientFinder` or similar if it helps.

Since we're moving more stuff from API into Core, I'd move the implementation (`PatientLoaderMetriportAPI`) to a `command` folder or similar, just because that's tied to a specific technology and it's not very domain material.

Reply:
renamed everything to PatientFinder and followed the PatientLoader framework. Moved stuff to command as well.